pkg/await/core_service_test.go

```diff
@@ -0,0 +1,290 @@
+package await
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+func Test_Core_Service(t *testing.T) {
+	tests := []struct {
+		description   string
+		do            func(services, endpoints chan watch.Event, settled chan struct{}, timeout chan time.Time)
+		expectedError error
+	}{
+		{
+			description: "Should succeed when Service is allocated an IP address and Endpoints target a Pod",
+			do: func(services, endpoints chan watch.Event, settled chan struct{}, timeout chan time.Time) {
+				// API server passes initialized service and endpoint objects back.
+				services <- watchAddedEvent(initializedService("default", "foo-4setj4y6"))
+				endpoints <- watchAddedEvent(initializedEndpoint("default", "foo-4setj4y6"))
+
+				// Mark endpoint objects as having settled. Success.
+				settled <- struct{}{}
+			},
+		},
+		{
+			description: "Should succeed if Endpoints have settled when timeout occurs",
+			do: func(services, endpoints chan watch.Event, settled chan struct{}, timeout chan time.Time) {
+				// API server passes initialized service back.
+				services <- watchAddedEvent(initializedService("default", "foo-4setj4y6"))
+
+				// Pass initialized endpoint objects.
+				endpoints <- watchAddedEvent(initializedEndpoint("default", "foo-4setj4y6"))
+
+				// DO NOT mark as settled. Instead, time out.
```

Comment:
This doesn't (to me) match whats in the description.

Reply:
This is correct. We're not passing `settled <- struct{}{}`. I'll reword though.